### ISSUE TITLE
MODX 3 - UX/UI/Functionality Changes

### ISSUE THREAD

**Issue by kolbykruger** [modxcms/revolution#13988]
### MODX 3 Changes

Below are some of the UX/UI/Functionality changes MODX3 needs. 

### UX / UI
**Files**

- Format file information into two columns (File Name and Path in the left; File Size, Last Accessed and Last Modified in the right). Reason: This shrinks the top space being used, allowing more room for the file's content to increase in size (especially on small screens).

**Chunks**

- Hide 'Tip' notification at the top. Not necessary unless it's the first chunk created. Reason: Allows more room for the chunk.

**Template Variables**

- Hide 'Tip' notification at the top. Not necessary unless it's the first chunk created. Reason: Allows more room for the TV.

**Snippets**

- Hide 'Tip' notification at the top. Not necessary unless it's the first chunk created. _Reason: Allows more room for the Snippets._

**Plugins**

- Hide 'Tip' notification at the top. Not necessary unless it's the first chunk created. _Reason: Allows more room for the Plugins._

**Categories for all Elements**

- Remove pagination. _Reason: A HUGE usability improvement to be able to scroll through all of the categories. The pagination is clunky and unnecessary._

**Toasts**

- Toasts (like the 'save successful' message) should display on the bottom right of the window. These should be a solid background color, with a maximum width of no larger than 300px. _Reason: More modern feel, better location that doesn't block functionality._

**Media Browser**

- More padding around the media browser items would allow for a more modern feel.
- Items have too many borders. Remove border wrapping the individual image, possibly make an off-white background to differentiate items

**Dashboard**

- Action buttons should stretch to full width (25%) - mentioned in a previous issue here:
- Resize and delete buttons need to appear/hide via a system setting
- Close the gap between the tray an the content area
- Scrollbars on windows are a hinderance to how the sidebar functions.
- Recently edited resources 'Resource's title' font size is off. It should be the same as the ID and then ellipsis if too long
- Recently edited resources buttons error out on click

**Edit Account**

- Move to two column layout
- Profile image should display in the right column (clipped by a circle for consistency)

**Manage -> Users**

- Include profile picture

### Functionality
**Extras**

- Most Popular -> Change to most popular for 20XX year. Overall most popular is too arbitrary.

**Site name**

- The site name needs to appear somewhere on the sidebar/tray. I would suggest adding it on the tray above the tabs (Resource/Elements/Files) as this will be open most of the time

### Navigation
**Content**
Remove 'Preview Site' link and add to overall navigation via 'Eyeball / View' icon. _Reason: This button is highly used, and hidden within the navigation. This should be a top level link as it is used frequently._

**Lower navigation**

- Reorder navigation so that the User profile is the last item. _Reason: Big circle on top of small circle's is imbalanced._
- Add system setting to 'toggle' the help icon (this is 99% unnecessary for clients, and mostly useful for developers)

### UX / UI
**Files**

- Format file information into two columns (File Name and Path in the left; File Size, Last Accessed and Last Modified in the right). Reason: This shrinks the top space being used, allowing more room for the file's content to increase in size (especially on small screens).

**Chunks**

- Hide 'Tip' notification at the top. Not necessary unless it's the first chunk created. Reason: Allows more room for the chunk.

**Template Variables**

- Hide 'Tip' notification at the top. Not necessary unless it's the first chunk created. Reason: Allows more room for the TV.

**Snippets**

- Hide 'Tip' notification at the top. Not necessary unless it's the first chunk created. _Reason: Allows more room for the Snippets._

**Plugins**

- Hide 'Tip' notification at the top. Not necessary unless it's the first chunk created. _Reason: Allows more room for the Plugins._

**Categories for all Elements**

- Remove pagination. _Reason: A HUGE usability improvement to be able to scroll through all of the categories. The pagination is clunky and unnecessary._

**Toasts**

- Toasts (like the 'save successful' message) should display on the bottom right of the window. These should be a solid background color, with a maximum width of no larger than 300px. _Reason: More modern feel, better location that doesn't block functionality._

**Media Browser**

- More padding around the media browser items would allow for a more modern feel.
- Items have too many borders. Remove border wrapping the individual image, possibly make an off-white background to differentiate items

**Dashboard**

- Action buttons should stretch to full width (25%) - mentioned in a previous issue here:
- Resize and delete buttons need to appear/hide via a system setting
- Close the gap between the tray an the content area
- Scrollbars on windows are a hinderance to how the sidebar functions.
- Recently edited resources 'Resource's title' font size is off. It should be the same as the ID and then ellipsis if too long
- Recently edited resources buttons error out on click

**Edit Account**

- Move to two column layout
- Profile image should display in the right column (clipped by a circle for consistency)

**Manage -> Users**

- Include profile picture

### Functionality
**Extras**

- Most Popular -> Change to most popular for 20XX year. Overall most popular is too arbitrary.

**Site name**

- The site name needs to appear somewhere on the sidebar/tray. I would suggest adding it on the tray above the tabs (Resource/Elements/Files) as this will be open most of the time

### Navigation
**Content**
Remove 'Preview Site' link and add to overall navigation via 'Eyeball / View' icon. _Reason: This button is highly used, and hidden within the navigation. This should be a top level link as it is used frequently._

**Lower navigation**

- Reorder navigation so that the User profile is the last item. _Reason: Big circle on top of small circle's is imbalanced._
- Add system setting to 'toggle' the help icon (this is 99% unnecessary for clients, and mostly useful for developers)

Let me know your thoughts!

**Comment by Ibochkarev** [comment]
With most proposals I agree. will be more convenient

**Comment by muzzwood** [comment]
Where is the pagination for element categories? 

Edit: Ah never mind, you mean the combobox on the element page.

I agree with everything you've said here.

**Comment by kolbykruger** [comment]
Yeah the combo box for selecting categories. It gets pretty clunky when you have 20+ categories, and you have to paginate through them to find the one you need.

**Comment by muzzwood** [comment]
There should still be a limit though I think. Perhaps just higher? 100?

**Comment by kolbykruger** [comment]
Yeah if they don't want to restructure it entirely, they could just allow us to adjust the amount shown. I think 100 is reasonable considering I will most likely never go over that. But having it paginate at 20 shows too little and I end up with 3-4 pages of categories that I have to sift through.

**Comment by sl1rt** [comment]
Hello! I am second time on GitHub, and don't know where to post my ideas, so I write it here))))
I am bad in English, sorry))

I think it's good to add 2 changes on manager:
1. Add checkbox - "Exclude resource from alias" for excluding resources name from default alias for friendly URLs setting.
2. Add richtext editor for "description" field of document if in settings active checkbox that activating this richtext editor for description)))
3. Template of "signupemail_message" - that sends when user change password doesn't work if I add it on some Context, system only use System setting value...

**Comment by cjholowatyj** [comment]
Great recommendations - every one of them! One thing I think MODx needs as well is:
**Typable (filtering) drop-down lists***...
...especially for the template TV! This would save so much time in my workflow because I know the names of the options I want to choose, just getting there can be slow and tedious some times...

**Comment by kolbykruger** [comment]
I 100% agree with that. If they made the global change for all dropdowns to have type searching, it increase my productivity 10-fold. It would also solve issues with dropdowns that are paginated, which goes against everything I believe in.

**Comment by ghost** [comment]
Oh, yeah! I can't count the number of times I've glanced at a dropdown and didn't notice that it was paginated, and spent hours trying to figure out why something I just did wasn't on the dropdown.

**Comment by JoshuaLuckers** [comment]
Feel free to contribute by submitting a PR😎👍

**Comment by SnowCreative** [comment]
> Yeah the combo box for selecting categories. It gets pretty clunky when you have 20+ categories, and you have to paginate through them to find the one you need.

What bothers me most is that you can't start typing to filter the categories list. Ability to type is inconsistent throughout the whole manager. For example, in some places, you can type to filter the list of resource templates; in others you can't. I think that ALL dropdown lists like this, whether or not they are paginated, should be filterable by typing.
